### PR TITLE
Add read_direct and write_direct functions

### PR DESCRIPTION
**Pull request by paulhfu**
To increase similarity to h5py

### CODE REVIEW

**Comment by paulhfu** [comment]
Hey @constantinpape I changed some stuff. Does this resolve your question? Not sure if I should also include all the slice generation and editing that is happening in h5py.

File: src/python/module/z5py/dataset.py [hunk 1]
```diff
@@ -515,6 +515,44 @@ def find_maximum_coordinates(self, dim):
...
+            dest_sel (slice array) selection of data to write to ``dest``. Defaults to the whole range of ``dest``.
+            source_sel (slice array) selection in dataset to read from. Defaults to the whole range of the dataset.
+        Spaces, defined by ``source_sel`` and ``dest_sel`` must be in the same size but dont need to have the same
+        offset
+        """
+        rTsl1 = lambda x, y: slice(x.__int__(), y.__int__())
+        if source_sel is None:
+            source_sel = tuple(map(rTsl1, [0 for s in self.shape], self.shape))
+        if dest_sel is None:
+            dest_sel = tuple(map(rTsl1, [0 for s in dest.shape], dest.shape))
```
Comment (constantinpape) [inline src/python/module/z5py/dataset.py]: I think I get what you are doing here, but this code seems way to complicated.
You could achieve the same with
``` python
if source_sel is None:
    source_sel = tuple(slice(0, sh) for sh in self.shape)
```
and similar for `dest_sel`.

File: src/python/module/z5py/dataset.py [hunk 2]
```diff
@@ -515,6 +515,44 @@ def find_maximum_coordinates(self, dim):
...
+            ``source``.
+            dest_sel (slice array) selection in dataset to write to. Defaults to the whole range of the dataset.
+        Spaces, defined by ``source_sel`` and ``dest_sel`` must be in the same size but dont need to have the same
+        offset
+        """
+        rTsl1 = lambda x, y: slice(x.__int__(), y.__int__())
+        if dest_sel is None:
+            dest_sel = tuple(map(rTsl1, [0 for s in source.shape], source.shape))
+        if source_sel is None:
+            source_sel = source_sel = tuple(map(rTsl1, [0 for s in self.shape], self.shape))
```
Comment (constantinpape) [inline src/python/module/z5py/dataset.py]: Please simplify, see comment above.

**Comment by constantinpape** [comment]
Good to merge now, thanks for the contribution!

**Pull request merged**